018-cli-2.0/018-cli-2.0.md

```diff
@@ -0,0 +1,431 @@
+- Appwrite Issue: NA
+
+## Summary
+
+[summary]: #summary
+
+Improve the Appwrite CLI, adding support for console SDK features, and an improved experience with cloud functions. 
+
+## Problem Statement (Step 1)
+
+[problem-statement]: #problem-statement
+
+**What problem are you trying to solve?**
+
+1. We cannot create and manage projects from the CLI
+2. Deploying functions is not very easy using the CLI
+3. Developers find it hard to create and test cloud functions locally.
+
+
+**What is the context or background in which this problem exists?**
+
+The CLI functionality is limited to a Server SDK which prevents us from creating and managing projects. There is also difficulty associated with the creation, packaging and deployment of functions. 
+
+
+**Once the proposal is implemented, how will the system change?**
+
+The CLI will behave more like the console SDK and provide options to create delete and manage all your projects without leaving the CLI. The CLI will also make it easier to create, test and deploy cloud functions.
+
+## Design proposal (Step 2)
+
+[design-proposal]: #design-proposal
+
+<!--
+This is the technical portion of the RFC. Explain the design in sufficient detail keeping in mind the following:
+
+- Its interaction with other parts of the system is clear
+- It is reasonably clear how the contribution would be implemented
+- Dependencies on libraries, tools, projects or work that isn't yet complete
+- New API routes that need to be created or modifications to the existing routes (if needed)
+- Any breaking changes and ways in which we can ensure backward compatibility.
+- Use Cases
+- Goals
+- Deliverables
+- Changes to documentation
+- Ways to scale the solution
+
+Ensure that you include examples, code-snippets etc. to allow the community to understand the proposed solution. **It would be best if the examples use naming conventions that you intend to use during the actual implementation so that changes can be suggested early on during the development.**
+
+Write your answer below.
+
+-->
+
+This refactor of the CLI will revolve around the following areas
+1. Migration of the CLI from a Server Side SDK to a Console SDK
+2. Easier deployment of Cloud Functions
+3. Easier creation of Cloud Functions
+
+
+## Migration of the CLI from a Server SDK to a Console SDK
+---
+ 
+Until now, the CLI was based off of our Server Side Swagger Spec. While this was a good start, it limits our functionality to a Server SDK. Ideally we want the CLI to be an alternative to the Console SDK ( aka the Appwrite Dashboard ). Here are some changes that we will need to make to adhere to the server spec.
+
+After the implementation of this RFC, the CLI will behave more like the Appwrite Console rather than a Server SDK. This accounts for a new form of Authentication that is similar to the way we handle logins in the Appwrite console. We use a cookie.
+
+We will introduce a new command `appwrite login` that will work in one of either ways.
```

Comment:
Do we need to pick one or the other? I understand that we should think to support OAuth login, but it's not a great CLI experience if a browser is required to login.